### ISSUE TITLE
Configure role hierarchy/inheritance

### ISSUE THREAD

**Issue by eloyekunle** [eloyekunle/PermissionsBundle#5]
The default Symfony roles system supports role inheritance. Still exploring how to implement this, but in a database-backed implementation, we can have a `weight` column, and roles with higher weights implement all permissions from all roles will lower weights.
Then the `RoleManager` can be extended to include Role hierarchy manipulation methods.

**Comment by anboo** [comment]
You can just override `security.role_hierarchy` service or create custom `RoleVoter` with maximum priority.

**Comment by eloyekunle** [comment]
Yes, but that will also be in addition to having a `weight` field. Then the `security.role_hierarchy` service that we're creating will use the weight to configure the hierarchy.

**Comment by eloyekunle** [comment]
Yes, but that will also be in addition to having a `weight` field. Then the `security.role_hierarchy` service that we're creating will use the weight to configure the hierarchy.
